### PR TITLE
Tile existence lemma

### PR DESCRIPTION
**Pull request by Parcly-Taxel**

### CODE REVIEW

**Comment by fpvandoorn** [comment]
Can you merge master, and remove `coe𝓓_injective`, and fix your proof?

**Comment by Parcly-Taxel** [comment]
> Can you merge master, and remove `coe𝓓_injective`, and fix your proof?

Done.

File: Carleson/GridStructure.lean [hunk 1]
```diff
@@ -101,19 +115,113 @@ postfix:max "ᵒ" => Grid.int
...
+    · exact (le_or_disjoint c).resolve_right nd
+  use j, ⟨mj, hj⟩, fun k ⟨hk₁, hk₂⟩ ↦ le_antisymm (hk₂ j mj) (hj k hk₁)
+
 open Classical in
 /-- If `i` is not a maximal element, this is the (unique) minimal element greater than i.
-Note, this is not a `SuccOrder`, since an element can be the successor of multiple other elements.
--/
-def succ (i : Grid X) : Grid X := if h : IsMax i then i else sorry
+This is not a `SuccOrder` since an element can be the successor of multiple other elements. -/
+def succ (i : Grid X) : Grid X := if h : IsMax i then i else Finset.choose (hp := exists_unique_succ i h)
```
[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Carleson/GridStructure.lean:143]: It doesn't change anything about the proofs.

File: Carleson/GridStructure.lean [hunk 2]
```diff
@@ -34,12 +34,13 @@ class GridStructure
   subset_topCube {i} : coeGrid i ⊆ coeGrid topCube
   Grid_subset_biUnion {i} : ∀ k ∈ Ico (-S) (s i), coeGrid i ⊆ ⋃ j ∈ s ⁻¹' {k}, coeGrid j
   fundamental_dyadic' {i j} : s i ≤ s j → coeGrid i ⊆ coeGrid j ∨ Disjoint (coeGrid i) (coeGrid j)
+  c_mem_Grid {i} : c i ∈ coeGrid i --2.0.10
```
[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Carleson/GridStructure.lean:37]: At the point where `GridStructure` is defined, `D` might be zero, in which case `ball_subset_Grid` degenerates to `∅ ⊆ coeGrid i` which clearly cannot prove `(coeGrid i).Nonempty`.

The blueprint still says that `c i` is contained in the cube `i`.

[the remark this answers is not in the document]
Comment (Parcly-Taxel) [reply Carleson/GridStructure.lean]: I found a way to please you, though. I've added a `NeZero D` instance to `GridStructure`, then later on I've synthesised this instance when `D = defaultD`.

[the remark this answers is not in the document]
Comment (fpvandoorn) [reply Carleson/GridStructure.lean]: My suggestion was to just prove `c_mem_Grid` under the assumption `[ProofData a q K σ₁ σ₂ F G]`, which we basically assume everywhere. But I'm also fine with this solution.

**Comment by fpvandoorn** [comment]
Thanks a lot!

**Pull request merged**